src/tables/avar.rs

```diff
@@ -1,43 +1,77 @@
-use otspec::types::*;
+use crate::table_delegate;
+use otspec::tables::avar::{avar as avar_ot, AxisValueMap, SegmentMap as SegmentMap_ot};
+use otspec::types::Tag;
 use otspec::Deserializer;
-use otspec_macros::tables;
 
 /// The 'avar' OpenType tag.
 pub const TAG: Tag = crate::tag!("avar");
 
-tables!(
-    AxisValueMap {
-        F2DOT14 fromCoordinate
-        F2DOT14 toCoordinate
-    }
-    SegmentMap {
-        Counted(AxisValueMap) axisValueMaps
+#[derive(Debug, PartialEq, Clone)]
+pub struct SegmentMap(pub Vec<(f32, f32)>);
```

Comment:
Okay, I think I understand some of what you were talking about in complaining about newtypes.

My general feeling is that the inner value of a newtype should not be public; making it public makes it impossible to uphold the invariants you're trying to use the type to represent.

In this case, keeping the inner type private means we could use the types defined in `otspec` directly, so we wouldn't need to go back and forth with the conversions, and would just need to convert when the type is created.